sw/device/tests/crypto/aes_gcm_testvectors.h

```diff
@@ -114,11 +116,33 @@ const aes_gcm_test_t kAesGcmTestvectors[3] = {
         .plaintext = kPlaintext,
         .aad_len = kAadLen,
         .aad = kAad,
+        .ciphertext = kCiphertext256,
+        .tag_len = 16,
         .tag =
             {// Tag = 324895b3d2f656e4fa2f8ce056137061
              0x32, 0x48, 0x95, 0xb3, 0xd2, 0xf6, 0x56, 0xe4, 0xfa, 0x2f, 0x8c,
              0xe0, 0x56, 0x13, 0x70, 0x61},
+    },
+
+    // 128-bit IV, 256-bit key, real message and aad, short tag
+    {
+        .key_len = ARRAYSIZE(kKey256),
+        .key = kKey256,
+        .iv_len = 16,
+        .iv =
+            {// IV = c58aded2e1bbecba8b16a5757e5475bd
+             0xc5, 0x8a, 0xde, 0xd2, 0xe1, 0xbb, 0xec, 0xba, 0x8b, 0x16, 0xa5,
+             0x75, 0x7e, 0x54, 0x75, 0xbd},
+        .plaintext_len = kPlaintextLen,
+        .plaintext = kPlaintext,
+        .aad_len = kAadLen,
+        .aad = kAad,
         .ciphertext = kCiphertext256,
+        .tag_len = 12,
+        .tag =
+            {// Tag = 324895b3d2f656e4fa2f8ce0
```

Comment:
Just for my own context: are there NIST test vectors for this configuration? It is always nice for me to be able to trace the test values into however they were generated out of band.

Reply:
I couldn't find any official test vectors, so I generated these quick-check ones using the pycryptodome Python library. Long-term, wycheproof has some AES-GCM test vectors we can use for more thorough/auditable testing.

Reply:
Sounds good. I think we are ok as long as the little-endian to big-endian conversion is always consistent at the API level.